2024-03-06 13:54:22,346: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-03-06 13:54:22,883: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-03-06 13:54:22,884: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,885: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-03-06 13:54:22,885: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,885: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,885: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,887: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,887: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,887: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,887: root: INFO: Current backtesting datetime 2023-12-16 09:30:00-05:00
2024-03-06 13:54:22,887: root: INFO: Current backtesting datetime 2023-12-18 08:30:00-05:00
2024-03-06 13:54:22,888: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,888: root: INFO: Current backtesting datetime 2023-12-18 09:30:00-05:00
2024-03-06 13:54:22,888: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,888: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,888: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,889: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,889: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,889: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,889: root: INFO: Current backtesting datetime 2023-12-19 09:30:00-05:00
2024-03-06 13:54:22,889: root: INFO: Current backtesting datetime 2023-12-19 08:30:00-05:00
2024-03-06 13:54:22,889: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,890: root: INFO: Current backtesting datetime 2023-12-19 09:30:00-05:00
2024-03-06 13:54:22,890: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,890: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,890: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,890: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,890: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,891: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,891: root: INFO: Current backtesting datetime 2023-12-20 09:30:00-05:00
2024-03-06 13:54:22,891: root: INFO: Current backtesting datetime 2023-12-20 08:30:00-05:00
2024-03-06 13:54:22,891: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,891: root: INFO: Current backtesting datetime 2023-12-20 09:30:00-05:00
2024-03-06 13:54:22,891: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,892: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,892: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,892: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,892: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,892: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,892: root: INFO: Current backtesting datetime 2023-12-21 09:30:00-05:00
2024-03-06 13:54:22,893: root: INFO: Current backtesting datetime 2023-12-21 08:30:00-05:00
2024-03-06 13:54:22,893: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,893: root: INFO: Current backtesting datetime 2023-12-21 09:30:00-05:00
2024-03-06 13:54:22,893: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,893: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,894: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,894: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,894: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,894: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,894: root: INFO: Current backtesting datetime 2023-12-22 09:30:00-05:00
2024-03-06 13:54:22,894: root: INFO: Current backtesting datetime 2023-12-22 08:30:00-05:00
2024-03-06 13:54:22,895: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,895: root: INFO: Current backtesting datetime 2023-12-22 09:30:00-05:00
2024-03-06 13:54:22,895: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,895: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,895: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,896: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,896: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,896: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,896: root: INFO: Current backtesting datetime 2023-12-23 09:30:00-05:00
2024-03-06 13:54:22,896: root: INFO: Current backtesting datetime 2023-12-26 08:30:00-05:00
2024-03-06 13:54:22,896: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,896: root: INFO: Current backtesting datetime 2023-12-26 09:30:00-05:00
2024-03-06 13:54:22,897: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,897: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,897: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,897: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,897: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,897: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,898: root: INFO: Current backtesting datetime 2023-12-27 09:30:00-05:00
2024-03-06 13:54:22,898: root: INFO: Current backtesting datetime 2023-12-27 08:30:00-05:00
2024-03-06 13:54:22,898: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,898: root: INFO: Current backtesting datetime 2023-12-27 09:30:00-05:00
2024-03-06 13:54:22,898: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,898: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,899: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,899: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,899: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,899: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,899: root: INFO: Current backtesting datetime 2023-12-28 09:30:00-05:00
2024-03-06 13:54:22,899: root: INFO: Current backtesting datetime 2023-12-28 08:30:00-05:00
2024-03-06 13:54:22,900: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,900: root: INFO: Current backtesting datetime 2023-12-28 09:30:00-05:00
2024-03-06 13:54:22,900: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,900: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,900: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,901: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,901: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,901: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,901: root: INFO: Current backtesting datetime 2023-12-29 09:30:00-05:00
2024-03-06 13:54:22,901: root: INFO: Current backtesting datetime 2023-12-29 08:30:00-05:00
2024-03-06 13:54:22,901: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,902: root: INFO: Current backtesting datetime 2023-12-29 09:30:00-05:00
2024-03-06 13:54:22,902: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,902: root: INFO: MLTrader : Executing the on_trading_iteration lifecycle method at 2024-03-06 13:54:22
2024-03-06 13:54:22,902: root: ERROR: Strategy.create_order() got an unexpected keyword argument 'type'
2024-03-06 13:54:22,902: root: ERROR: Traceback (most recent call last):
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 808, in run
    self._run_trading_session()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 759, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 283, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 307, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 86, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/strategies/strategy_executor.py", line 377, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Session 5/venv/Programming_Project/lib/python3.12/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nicolas/PycharmProjects/Programming_Project/Trading bot.py", line 18, in on_trading_iteration
    order = self.create_order(
            ^^^^^^^^^^^^^^^^^^
TypeError: Strategy.create_order() got an unexpected keyword argument 'type'

2024-03-06 13:54:22,902: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-03-06 13:54:22,903: root: INFO: MLTrader : Sleeping for 86400 seconds
2024-03-06 13:54:22,903: root: INFO: Current backtesting datetime 2023-12-30 09:30:00-05:00
2024-03-06 13:54:22,903: root: INFO: Current backtesting datetime 2024-01-02 08:30:00-05:00
2024-03-06 13:54:22,903: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-03-06 13:54:22,903: root: INFO: Current backtesting datetime 2024-01-02 09:30:00-05:00
2024-03-06 13:54:22,903: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-03-06 13:54:22,904: root: INFO: Current backtesting datetime 2024-01-02 15:55:00-05:00
2024-03-06 13:54:22,904: root: INFO: MLTrader : Executing the before_market_closes lifecycle method
2024-03-06 13:54:22,904: root: INFO: Current backtesting datetime 2024-01-02 16:00:00-05:00
2024-03-06 13:54:22,904: root: INFO: MLTrader : Executing the after_market_closes lifecycle method
2024-03-06 13:54:22,904: root: INFO: MLTrader : Executing the on_strategy_end lifecycle method
2024-03-06 13:54:22,905: backtest_stats: INFO: Backtest took 0:00:00.559159 for a speed of 0.000
2024-03-06 13:54:22,905: root: WARNING: Cannot plot returns because the strategy returns are missing
2024-03-06 13:54:22,998: root: WARNING: Cannot create a tearsheet because the strategy returns are missing
